Lesson6_subtree.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+Subtree of Another Tree
+Given two non-empty binary trees s and t,
+check whether tree t has exactly the same structure and node values with a subtree of s.
+A subtree of s is a tree that consists of a node in s and all of that node's descendants.
+
+*/
+#include<iostream>
+using namespace std;
+struct tree
+{
+	char data;
+	tree* left,*right;
+	tree* createNode(char d)
+	{
+		tree* p=new tree;
+		p->data=d;
+		p->left=NULL;
+		p->right=NULL;
+	}
+	int compareTree(tree* S,tree* T)
+	{
+		if((S==NULL)&&(T==NULL)) return 1;
+		if((S==NULL) || (T==NULL))return 0;
+		if((S->data==T->data) && compareTree(S->left,T->left) && compareTree(S->right,T->right))
+			return 1;
+	}
+	int checkIfSubtree(tree* S,tree* T)
+	{
+		if(S==NULL) return 0;
+		if(T==NULL) return 1;
+		if( compareTree(S,T)==1) return 1;
+		return (checkIfSubtree(S->left,T) || checkIfSubtree(S->right,T));
+	}
+	void inorder(tree* root)
+	{
+		if(root)
+		{
+			inorder(root->left);
+			cout<<root->data<<"--";
+			inorder(root->right);
+		}
+	}
+}obj;
+
+/*TEST CASES*/
+bool t1()
```

Comment:
Naming these test cases better would help your interviewer (and other developers you work with in the future!) to understand what it is you are testing in the first place. For example, t1 could be:
testBothEmptyReturnsTrue or something

I see you have it as a comment and thats good, but when you run your tests through a tool, or through a build system which is very likely when you have a job, all that it will show is the error message and "t1 failed". It would be a lot more helpful if it sayd "testBothEmptyReturnsTrue failed" because you know exactly what the problem is just from the test name

Otherwise good job with the implementation and tests!

Reply:
Thanks , I will make the required changes.